assert/assertion_compare_legacy.go

```diff
@@ -0,0 +1,11 @@
+// +build !go1.17
```

Comment:
Could you please add TODO comments in all these version-dependent stating that we can merge them back in once we're done supporting Go 1.17?